frontend/libs/ui/src/components/DataTable/DataTable.tsx

```diff
@@ -65,12 +67,12 @@ export class DataTableColumnProvider<T extends object> {
                 </Button>
               )}
               {this.props.onEditClick && (
-                <Button onClick={() => this.props.onEditClick && this.props.onEditClick(row)} color="warning">
+                <Button className='ml-1' onClick={() => this.props.onEditClick && this.props.onEditClick(row)} color="warning">
                   <Icon name="pen" />
                 </Button>
               )}
-              {this.props.onDeleteClick && (
-                <Button onClick={() => this._onDeleteClick(row)} color="danger">
+              {this.props.onDeleteClick && this.props.featureName !== 'user' && (
+                <Button className='ml-1' onClick={() => this._onDeleteClick(row)} color="danger">
```

Comment:
The `featureName` is an implementation detail.  
If we make cases in this component that depend on a magic string like 'user', it can make the component fragile.  
Can we solve this in a way that keeps the `DataTable` from knowing implementation details about the component that's using it?

`this.props.onDeleteClick` is controlled by the other component. Can custom functionality be put there instead?

Reply:
Are you trying to hide the button on the `DataTable` on the user editor component?

Reply:
If we remove the default `deleteClick` variable here, will it have the same effect?

https://github.com/savantly-net/sprout-platform/blob/master/frontend/libs/ui/src/components/DataTable/DataTable.tsx#L33

Reply:
actually, I am not very sure about deleteClick, Here I am trying to hide the button. I will remove `this.props.featureName !== 'user'` and find a better solution here.